### ISSUE TITLE
How to log the evaluator_results for MultipleNegativesRankingLoss training

### ISSUE THREAD

**Issue by FadelMF** [UKPLab/sentence-transformers#2179]
I want to visualize loss/accuracy in model.fit().
Tried evaluator using
```
# Data in list dict.
eval_train_dataset = [
   {
     "text": ["Sentence 1", "Sentence 2", "Sentence 3", "Sentence 4", "Sentence 5"],
     "summary": ["Sentence 2", "Sentence 3", "Sentence 4"]
    },
     ...
]
eval_sample_dataset = [
   {
     "text": ["Sentence 1", "Sentence 2", "Sentence 3", "Sentence 4", "Sentence 5"],
     "summary": ["Sentence 2", "Sentence 3", "Sentence 4"]
    },
     ...
]

eval_trains = []
for idx, data in eval_train_dataset.iterrows():
    print("Total data train sekarang:", idx, end="\r")
    eval_trains.append(InputExample(texts=[data["text"], data["summary"]]))
eval_samples = []
for idx, data in eval_train_dataset.iterrows():
    print("Total data train sekarang:", idx, end="\r")
    eval_samples .append(InputExample(texts=[data["text"], data["summary"]]))

EmbeddingSimilarityEvaluator.from_input_examples(
    eval_samples, write_csv=False
)
train_dataloader = DataLoader(eval_trains, batch_size=8)
train_loss = losses.MultipleNegativesRankingLoss(model=model)

epochs = 10
warmup_steps = int(len(train_dataloader) * epochs * 0.1)

model.fit(
    [(train_dataloader, train_loss)],
    epochs=epochs,
    warmup_steps=warmup_steps,
    max_grad_norm=1,
    evaluator=evaluator,
    evaluation_steps=100,
    optimizer_class=torch.optim.AdamW,
    optimizer_params={
        "lr": 3e-5
    },
    weight_decay= 0.0001,
    show_progress_bar=True
)
```

But it gives an error like this:
```
PearsonRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.
warnings.warn(PearsonRConstantInputWarning())
```

Please help me.
